lightkurve/utils.py

```diff
@@ -484,6 +484,17 @@ def detect_filetype(header):
     try:
         # use `telescop` keyword to determine mission
         # and `creator` to determine tpf or lc
+
+        # Non-pipeline file types from e.g. lightkurve .interact()
+        if "SOURCE" in header.keys():
+            # Could be a custom lightcurve
+            if header['SOURCE'] == 'lightkurve interact':
+                if header['mission'].lower() == 'tess':
+                    return 'TessLightCurveFile'
+                elif header['mission'].lower()[0] == 'k':
+                    return 'KeplerLightCurveFile'
```

Comment:
Do we need lines 488-495 here?  I think the existing lines that follow (497-517) will already identify an interact-generated file correctly (based on the TELESCOP and CREATOR keywords).  If that's true, then let's get rid of lines 488-495, because less code is better!

Reply:
The interact-generated fits files do not have the `TELESCOP` keyword, unfortunately.  We could add this header in the `to_fits()` method, but any existing `interact()`-generated fits files would not be readable.

Reply:
I don't think that's true, e.g. `tests/data/test_K2_interact_generated_custom-lc.fits` shows `TELESCOP= 'KEPLER  '` in the header.  Do you want to have a go to see if it all works without this extra code?

Reply:
TESS interact-generated fits files do not populate the `TELESCOP` keyword:

`TELESCOP=  / telescope`

I will populate this metadata in `to_fits()` to eliminate the extra code.

Reply:
Thought: rather than lots of if/then logic, we could define a nested dictionary here and use keys to get the filetype string: 

`lookup_table = {'tess':{'to_fits':'TESSLightCurveFile', 'targetpixel':'TESSLightCurveFile'}, 'kepler':[...]}`